pkg/translator/prometheusremotewrite/helper_v2.go

```diff
@@ -0,0 +1,68 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package prometheusremotewrite // import "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/translator/prometheusremotewrite"
+
+import (
+	"github.com/prometheus/common/model"
+	writev2 "github.com/prometheus/prometheus/prompb/io/prometheus/write/v2"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	conventions "go.opentelemetry.io/otel/semconv/v1.25.0"
+
+	prometheustranslator "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/translator/prometheus"
+)
+
+// addResourceTargetInfoV2 converts the resource to the target info metric.
+func addResourceTargetInfoV2(resource pcommon.Resource, settings Settings, timestamp pcommon.Timestamp, converter *prometheusConverterV2) {
+	if settings.DisableTargetInfo || timestamp == 0 {
+		return
+	}
+
+	attributes := resource.Attributes()
+	identifyingAttrs := []string{
+		string(conventions.ServiceNamespaceKey),
+		string(conventions.ServiceNameKey),
+		string(conventions.ServiceInstanceIDKey),
+	}
+	nonIdentifyingAttrsCount := attributes.Len()
+	for _, a := range identifyingAttrs {
+		_, haveAttr := attributes.Get(a)
+		if haveAttr {
+			nonIdentifyingAttrsCount--
+		}
+	}
+	if nonIdentifyingAttrsCount == 0 {
+		// If we only have job + instance, then target_info isn't useful, so don't add it.
+		return
+	}
+
+	name := prometheustranslator.TargetInfoMetricName
+	if len(settings.Namespace) > 0 {
+		// TODO what to do with this in case of full utf-8 support?
+		name = settings.Namespace + "_" + name
```

Comment:
we have functions like "BuildCompliantMetricName" that take care of this

Reply:
But we still want to use library functions whenever we perform this kind of conversion. We have a lot of issues stemming from people reimplementing translation logic in many different places many different ways. If the existing API is not sufficient, we need to fix the APIs so they can work in more cases. For instance, we could add a new accessor to https://github.com/prometheus/otlptranslator/blob/1aca92dfe1eac8e7fbcfc44064a5a066aee81ce5/metric_namer.go#L110 that doesn't require the underlying metric type.

Reply:
sounds good